docs/50_Writable_Views.md

```diff
@@ -0,0 +1,521 @@
+    },
+    {
+      "name": "date",
+      "type": "Date",
+      "required": false,
+      "nullable": false
+    },
+    {
+      "name": "totalAmount",
+      "type": "number",
+      "required": false,
+      "nullable": false
+    },
+    {
+      "name": "items",
+      "type": "Array",
+      "items": {
+        "type": "string"
+      },
+      "description": "The item to deliver to the customer",
+      "required": true,
+      "nullable": false
+    },
+    {
+      "name": "rider",
+      "type": "RawObject",
+      "schema": {
+        "properties": {
+          "value": {
+            "type": "string",
+            "__mia_configuration": {
+              "type": "ObjectId"
+            }
+          },
+          "label": {
+            "type": "string"
+          }
+        }
+      },
+      "additionalProperties": false,
+      "required": true,
+      "nullable": false
+    }
+  ],
+  "indexes": [
+    {
+      "name": "_id",
+      "type": "normal",
+      "unique": true,
+      "fields": [
+        {
+          "name": "_id",
+          "order": 1
+        }
+      ]
+    }
+  ]
+}
+```
+
+```json title=View Definition
+{
+  "name": "orders-details",
+  "source": "orders",        // source collection
+  "type": "view",
+  "enableLookup": true,      // enable the lookup feature
+  "pipeline": [
+    {
+      "$lookup": {
+        "from": "riders",    // lookup collection
+        "localField": "riderId",
+        "foreignField": "_id",
+        "as": "rider",
+        "pipeline": [
+          {
+            "$match": {
+              "__STATE__": "PUBLIC"
+            }
+          },
+          {
+            "$project": {
+              "_id": 0,      // to hide the _id field in the view
+              "value": {
+                "$toObjectId": "$_id"     // foreign key
+              },
+              "label": {
+                "$toString": {
+                  "$concat": ["$firstName", " ", "$lastName"]     // the aggregated value
+                }
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "$unwind": {
+        "path": "$rider",
+        "preserveNullAndEmptyArrays": true
+      }
+    },
+    {
+      "$project": {
+        "riderId": 0
+      }
+    }
+  ]
+}
+```
+
+Configuring the CRUD Service with the above collection and view definitions, and calling
+the view `GET` endpoint would result in the following element:
+
+[//]: # (TODO: add curl)
+
+```json
+[
+  {
+    "_id": "6554e9eedf0efad9177c8212",
+    "date": "2023-11-17T14:17:08.770Z",
+    "totalAmount": 12.5,
+    "items": [
+    ],
+    "rider": {
+      "value": "65576e773a7e5c988fee1084",
+      "label": "Frodo Baggins"
+    },
+    "__STATE__": "PUBLIC"
+  }
+]
+```
+
+As it can be observed, the `riderId` has been replaced with an object representing the reference to the  record on the lookup collection. The application can now employ `PATCH` operations to change the reference on the underlying 
```

Comment:
```suggestion
As it can be observed, the `riderId` has been replaced with an object representing the reference to the  record on the lookup collection. The application can now employ `PATCH` operations to change the reference on the underlying collection
```